loopback/loopback.go

```diff
@@ -0,0 +1,539 @@
+func (n *Node) Setattr(ctx context.Context,
+	req *fuse.SetattrRequest, resp *fuse.SetattrResponse) (err error) {
+	defer func() { log.Printf("%s.Setattr(): error=%v", n.realPath, err) }()
+	if req.Valid.Size() {
+		if err = syscall.Truncate(n.realPath, int64(req.Size)); err != nil {
+			return nil
+		}
+	}
+
+	if req.Valid.Mtime() {
+		var tvs [2]syscall.Timeval
+		if !req.Valid.Atime() {
+			tvs[0] = tToTv(time.Now())
+		} else {
+			tvs[0] = tToTv(req.Atime)
+		}
+		tvs[1] = tToTv(req.Mtime)
+	}
+
+	if req.Valid.Handle() {
+		// panic("no idea what this is for")
+	}
+
+	if req.Valid.Mode() {
+		if err = os.Chmod(n.realPath, req.Mode); err != nil {
+			return err
+		}
+	}
+
+	if req.Valid.Uid() || req.Valid.Gid() {
+		if req.Valid.Uid() && req.Valid.Gid() {
+			if err = os.Chown(n.realPath, int(req.Uid), int(req.Gid)); err != nil {
+				return err
+			}
+		}
+		fi, err := os.Stat(n.realPath)
+		if err != nil {
+			return err
+		}
+		s := fi.Sys().(*syscall.Stat_t)
+		if req.Valid.Uid() {
+			if err = os.Chown(n.realPath, int(req.Uid), int(s.Gid)); err != nil {
+				return err
+			}
+		} else {
+			if err = os.Chown(n.realPath, int(s.Uid), int(req.Gid)); err != nil {
+				return err
+			}
+		}
+	}
+
+	if req.Valid.Flags() {
+		if err = syscall.Chflags(n.realPath, int(req.Flags)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+var _ fs.NodeRenamer = (*Node)(nil)
+
+// Rename implements fs.NodeRenamer interface for *Node
+func (n *Node) Rename(ctx context.Context,
+	req *fuse.RenameRequest, newDir fs.Node) (err error) {
+	np := filepath.Join(newDir.(*Node).realPath, req.NewName)
+	op := filepath.Join(n.realPath, req.OldName)
+	defer func() {
+		log.Printf("%s.Rename(%s->%s): error=%v",
+			n.realPath, op, np, err)
+	}()
+	return os.Rename(op, np)
+}
+
+// Handle represent an open file or directory
+type Handle struct {
+	fs       *FS
+	reopener func() (*os.File, error)
+
+	f *os.File
+}
+
+var _ fs.HandleFlusher = (*Handle)(nil)
+
+// Flush implements fs.HandleFlusher interface for *Handle
+func (h *Handle) Flush(ctx context.Context,
+	req *fuse.FlushRequest) (err error) {
+	defer func() { log.Printf("Handle(%s).Flush(): error=%v", h.f.Name(), err) }()
+	return h.f.Sync()
+}
+
+var _ fs.HandleReadAller = (*Handle)(nil)
+
+// ReadAll implements fs.HandleReadAller interface for *Handle
+func (h *Handle) ReadAll(ctx context.Context) (d []byte, err error) {
+	defer func() {
+		log.Printf("Handle(%s).ReadAll(): error=%v",
+			h.f.Name(), err)
+	}()
+	return ioutil.ReadAll(h.f)
+}
+
+var _ fs.HandleReadDirAller = (*Handle)(nil)
+
+// ReadDirAll implements fs.HandleReadDirAller interface for *Handle
+func (h *Handle) ReadDirAll(ctx context.Context) (
+	dirs []fuse.Dirent, err error) {
+	defer func() {
+		log.Printf("Handle(%s).ReadDirAll(): %#+v error=%v",
+			h.f.Name(), dirs, err)
+	}()
+	fis, err := h.f.Readdir(0)
+	if err != nil {
+		return nil, err
+	}
+	if err = h.f.Close(); err != nil {
```

Comment:
Can you add a comment on why this is necessary?  Is it because `Readdir` won't return anything on the second call?